app/src/pages/ExploreDataPage.tsx

```diff
@@ -6,18 +6,49 @@ import FormControl from "@material-ui/core/FormControl";
 import MenuItem from "@material-ui/core/MenuItem";
 import DemoReport from "../features/reports/DemoReport";
 import TellMeAboutReport from "../features/reports/TellMeAboutReport";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogContentText from "@material-ui/core/DialogContentText";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import IconButton from "@material-ui/core/IconButton";
+import ShareIcon from "@material-ui/icons/Share";
+import { getMadLibPhraseText } from "../utils/MadLibs";
+
 import {
   MADLIB_LIST,
   MadLib,
   PhraseSegment,
   PhraseSelections,
 } from "../utils/MadLibs";
 import styles from "./ExploreDataPage.module.scss";
+import {
+  clearSearchParams,
+  buildMadLibSelectionParams,
+  MADLIB_PHRASE,
+  MADLIB_SELECTIONS,
+  useSearchParams,
+} from "../utils/urlutils";
 
 function ExploreDataPage() {
-  const [phraseIndex, setPhraseIndex] = useState(0);
+  const [shareModalOpen, setShareModalOpen] = React.useState(false);
+  const params = useSearchParams();
+  useEffect(() => {
+    clearSearchParams([MADLIB_PHRASE, MADLIB_SELECTIONS]);
+  }, []);
+  const [phraseIndex, setPhraseIndex] = useState<number>(
+    Number(params[MADLIB_PHRASE]) | 0
+  );
+
+  let defaultValuesWithOverrides = MADLIB_LIST[phraseIndex].defaultSelections;
+  if (params[MADLIB_SELECTIONS]) {
+    params[MADLIB_SELECTIONS].split(",").forEach((override) => {
+      console.log("override", override);
+      const [key, value] = override.split(":");
+      defaultValuesWithOverrides[Number(key)] = Number(value);
```

Comment:
It would be nice to validate that this is an acceptable value. I tried plugging some random thing into one of the overrides and it results in that dropdown being unpopulated. (I consider this not too important because I don't know how likely it is someone will be messing with the url, but it's a nice check just to make sure rogue urls don't break the UI)

Reply:
oh good call!